src/System.Net.Security/src/System/Net/Unix/SSPISecureChannelType.cs

```diff
@@ -0,0 +1,290 @@
+        {
+            return HandshakeInternal(credential, ref context, inputBuffer, outputBuffer, true, remoteCertRequired);
+        }
+
+        public SecurityStatus InitializeSecurityContext(ref SafeFreeCredentials credential, ref SafeDeleteContext context,
+            string targetName, SecurityBuffer inputBuffer, SecurityBuffer outputBuffer)
+        {        
+            return HandshakeInternal(credential, ref context, inputBuffer, outputBuffer, false, false);
+        }
+
+        public SecurityStatus InitializeSecurityContext(SafeFreeCredentials credential, ref SafeDeleteContext context, string targetName, SecurityBuffer[] inputBuffers, SecurityBuffer outputBuffer)
+        {          
+            Debug.Assert(inputBuffers.Length == 2);
+            Debug.Assert(inputBuffers[1].token == null);
+            return HandshakeInternal(credential, ref context, inputBuffers[0], outputBuffer, false, false);
+        }
+
+        public SecurityStatus EncryptMessage(SafeDeleteContext securityContext, byte[] buffer, int size, int headerSize, int trailerSize, out int resultSize)
+        {
+            return EncryptDecryptHelper(securityContext, buffer, 0, size, headerSize, trailerSize, true,
+                out resultSize);
+        }
+
+        public SecurityStatus DecryptMessage(SafeDeleteContext securityContext, byte[] buffer, ref int offset, ref int count)
+        {
+            int resultSize;
+            SecurityStatus retVal = EncryptDecryptHelper(securityContext, buffer, offset, count, 0, 0, false, out resultSize);
+            if (SecurityStatus.OK == retVal)
+            {
+                count = resultSize;
+            }
+            return retVal;
+        }
+
+        public int QueryContextChannelBinding(SafeDeleteContext phContext, ChannelBindingKind attribute,
+            out SafeFreeContextBufferChannelBinding refHandle)
+        {
+            throw NotImplemented.ByDesignWithMessage(SR.net_MethodNotImplementedException);
+        }
+
+        public int QueryContextStreamSizes(SafeDeleteContext securityContext, out StreamSizes streamSizes)
+        {
+            streamSizes = null;
+            try
+            {
+                streamSizes = new StreamSizes(Interop.libssl.SslSizes.HEADER_SIZE, Interop.libssl.SslSizes.TRAILER_SIZE, Interop.libssl.SslSizes.SSL3_RT_MAX_PLAIN_LENGTH);
+                return 0;
+            }
+            catch
+            {              
+                return -1;
+            }
+        }
+
+        public int QueryContextConnectionInfo(SafeDeleteContext securityContext, out SslConnectionInfo connectionInfo)
+        {
+            bool gotReference = false;
+            connectionInfo = null;
+            try
+            {
+                securityContext.DangerousAddRef(ref gotReference);
+                Interop.libssl.SSL_CIPHER cipher = Interop.OpenSsl.GetConnectionInfo(securityContext.DangerousGetHandle());
```

Comment:
libssl does not require the pointer to be freed. We could still wrap it inside a SafeHandle but just want to check if the usage is appropriate